### PR TITLE
Issues-2534 - language fallback fix

### LINKED ISSUE #2534: Italian language with Google TTS
I'm running Mycroft on Manjaro Linux, with Italian language.
I tried both the "British male" and "American male" voices, and they do not speak Italian at all. So I decided to try the Google Voice.
That way, no sound is emitted. Whenever Mycroft tries to speak, I see this error in logs (file audio.log):

```
2020-04-13 10:45:39.632 | INFO     | 195922 | mycroft.audio.speech:mute_and_speak:127 | Speak: Va uno spettacolo
2020-04-13 10:45:40.070 | ERROR    | 195922 | mycroft.audio.speech:handle_speak:99 | Error in mute_and_speak
Traceback (most recent call last):
  File "/home/luke/git/mycroft-core/mycroft/audio/speech.py", line 95, in handle_speak
    mute_and_speak(chunk, ident, listen)
  File "/home/luke/git/mycroft-core/mycroft/audio/speech.py", line 129, in mute_and_speak
    tts.execute(utterance, ident, listen)
  File "/home/luke/git/mycroft-core/mycroft/tts/tts.py", line 337, in execute
    wav_file, phonemes = self.get_tts(sentence, wav_file)
  File "/home/luke/git/mycroft-core/mycroft/tts/google_tts.py", line 35, in get_tts
    tts = gTTS(text=sentence, lang=self.lang)
  File "/home/luke/git/mycroft-core/.venv/lib/python3.8/site-packages/gtts/tts.py", line 121, in __init__
    raise ValueError("Language not supported: %s" % lang)
ValueError: Language not supported: it-it
```

The problem is that Google TTS can handle "it", but it cannot handle "it-it".

I will release a PR shortly for fixing this. (So far, Google Voice seems my only option for Italian language, so I really need that).

### PR DESCRIPTION
**Pull request by luca-vercelli**
## Description
Fixes #2534

## How to test
Run mycroft-core with language it-it, and use Google Voice as TTS

## Contributor license agreement signed?
CLA

### CODE REVIEW

**Comment by devs-mycroft** [comment]
Hello, @luca-vercelli, thank you for helping with the Mycroft project! We welcome everyone
into the community and greatly appreciate your help as we work to build an AI
for Everyone.

To protect yourself, the project, and users of Mycroft technologies we require
a Contributor Licensing Agreement (CLA) before accepting any code
contribution. This agreement makes it crystal clear that along with your
code you are offering a license to use it within the confines of this project.
You retain ownership of the code, this is just a license.

Please visit https://mycroft.ai/cla to initiate this one-time signing. Thank
you!

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2535))

File: mycroft/tts/google_tts.py [hunk 1]
```diff
@@ -32,6 +33,12 @@ def get_tts(self, sentence, wav_file):
         Returns:
             Tuple ((str) written file, None)
         """
+        langs = tts_langs()
+        if self.lang.lower() not in langs:
+            if self.lang[:2].lower() in langs:
+                self.lang = self.lang[:2]
+            else:
+                raise ValueError("Language not supported by gTTS: %s" % lang)
```
Comment (forslund) [inline mycroft/tts/google_tts.py:41]: This should be `self.lang` right?

Also we try to use the `.format()` for string formatting so can you change it to
`raise ValueError('Language not supported by gTTS: {}'.format(self.lang)`

Comment (luca-vercelli) [reply mycroft/tts/google_tts.py]: just done.

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2535))

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2535))

**Comment by forslund** [comment]
Tested it with swedish and works great!

I had one thought when viewing it offline, it might be cleaner to have the language change done in the `__init__()` instead of every time a sentence is parsed:

The `__init__()` would become
```python
    def __init__(self, lang, config):                                           
        super(GoogleTTS, self).__init__(lang, config, GoogleTTSValidator(       
            self), 'mp3')                                                       
        langs = tts_langs()                                                     
        if self.lang.lower() not in langs:                                      
            if self.lang[:2].lower() in langs:                                  
                self.lang = self.lang[:2]                                       
            else:                                                               
                raise ValueError("Language not supported by gTTS: {}"           
                                 .format(self.lang))
```

What do you think?

**Comment by luca-vercelli** [comment]
I agree, you are right.
Moreover, I have just seen a method "validate_lang" a few lines below... with a "TODO" annotation... maybe that's the right place to put the code?

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2535))

**Comment by forslund** [comment]
Hmmm, not sure really, the "validate" in the method name indicates that it is a check and should not do any modifications.

I would split it up and do the modification in the `__init__()` or set to None if no lang can be determined. Then raise the exception in the validate_lang method.

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2535))

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Failed ([Results](https://reports.mycroft.ai/core/PR-2535))

**Comment by krisgesling** [comment]
CLA received, but looks like this may still be in progress.

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2535))

**Comment by forslund** [comment]
I think this looks good, I'll rebase it and squash the pep8 / typo fixes into their parent commits to make the commit history cleaner and then I'll merge it.

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2535))

**Pull request merged**